### PR TITLE
Fix issues tab 

### LINKED ISSUE #5358: No issues notification circle for Entity problems such as missing description or subject item id
We don't show a notification circle number next to the issues tab in the Schema view for Entity problems that result in the schema being incomplete.

### To Reproduce
1. Use Clipboard importer in OpenRefine and copy and paste the following simple 2 row example dataset

| A | B | C |
| - | - | - |
| 1 | 2 | 3 |
| 4 | 5 | 6 |

2. Click **Create Project**
3. Click on **Wikidata Extension**  -> **Edit Wikibase Schema**
4. Click **Add item** button
5. On Terms click **add term** link
6. From dropdown change Alias to Description but do not type anything into boxes.
7. On top header, click on **Issues** tab to ensure there are 2 issues reported for the empty entity.

### Current Results

No notification circle is shown on the Issues tab for the count of 2 issues.

### Expected Behavior

The issues tab should show the notification circle with number 2 signifying 2 issues are reported under the tab.

### Screenshots
No number is shown on the issues tab 

image

However, when the issues tab is clicked, you can see 2 entity issues do exist

image

I expected to see a number 2 next to the issues tab as in this example:

image

### Versions
 - Operating System: Windows 11
 - Browser Version: Edge latest
 - JRE or JDK Version: embedded in OpenRefine
 - OpenRefine: Version 3.7-SNAPSHOT [ea70840]

### PR DESCRIPTION
**Pull request by SoniaSun810**
Fixes #5358

Changes proposed in this pull request:
- Function `SchemaAlignment.preview` and `SchemaAlignment._updateWarnings` are modified in the file `extensions/wikibase/module/scripts/schema-alignment.js`;
- We discovered that both errors and warnings could contribute to the overall issue count. As a result, we have relocated the logic for "updating the counts in the issues tab" to the `SchemaAlignment.preview` function, departing from its previous location in the `SchemaAlignment._updateWarnings` function.

**Further question:**
Would it be beneficial to add new test cases for the `wikibase-schema` component within the frontend scope?

### CODE REVIEW

**Review by wetneb — COMMENTED** [review]
Amazing, thanks a lot! I really like that you introduced Cypress tests for this too.

Overall, I wonder if the restructuring you introduced is really necessary? There should not be any need to sum the number of warnings and the number of errors (because they cannot be non-zero simultaneously.

Could something like this work instead (without needing to make any changes to `_updateWarnings`)?
```javascript
if (data.warnings) {
     self._updateWarnings(data.warnings, data.nb_warnings);
} else {
    self._updateWarnings([], data.errors ? data.errors.length : 0);
}
```

File: extensions/wikibase/module/scripts/schema-alignment.js [hunk 1]
```diff
@@ -1689,11 +1690,21 @@ SchemaAlignment.preview = function() {
         self.updateNbEdits(data["edit_count"]);
       }
 
-      if (data.warnings) {
-          self._updateWarnings(data.warnings, data.nb_warnings);
-      } else {
-          self._updateWarnings([], 0);
-      }
+      // update the counts in the issues tab
+     var numWarnings = data.warnings ? data.warnings.length : 0;
```
[the remark this answers is not in the document]
Comment (SoniaSun810) [reply extensions/wikibase/module/scripts/schema-alignment.js:1694]: Thank you for your suggestion! I agree that `data.nb_warnings` is more precise. I will modify it as soon as possible! 😄

**Comment by SoniaSun810** [comment]
> Overall, I wonder if the restructuring you introduced is really necessary? There should not be any need to sum the number of warnings and the number of errors (because they cannot be non-zero simultaneously.
> 
> Could something like this work instead (without needing to make any changes to `_updateWarnings`)?

Thank you for your question. The main rationale behind the restructuring was to ensure that the issue count only updates once when a change occurs. Prior to this change, the issue count would also update in cases where `data.warnings` was "undefined". This happened due to the invocation of the `_updateWarning` function, which led to the display number adjacent to the issue tab being modified again.

**Review by wetneb — APPROVED** [review]
Thank you again!

**Pull request merged**